### DISCUSSION TITLE
Why is PROJECT_SOVERSION bumped for patch releases?

### DISCUSSION THREAD

**Discussion by scpeters** [open-source-parsers/jsoncpp#1384] [General]
Hi, I noticed that the `PROJECT_SOVERSION` cmake variable is often bumped with patch releases, such as in https://github.com/open-source-parsers/jsoncpp/pull/1351#discussion_r789600084. According to the [semantic versioning statement](https://github.com/open-source-parsers/jsoncpp#a-note-on-backward-compatibility) for this project, "Major versions maintain binary-compatibility." I'm not an expert on the reasons to update an SOVERSION variable, but I have observed that this causes on macOS that requires binaries to be rebuilt. If we are maintaining binary compatibility, it's a shame to invalidate binaries though bumping this variable (if that is what's happening).

* https://github.com/Homebrew/homebrew-core/pull/92195
* https://github.com/osrf/homebrew-simulation/issues/1780

**Comment by baylesj** [comment]
I think we have been overly aggressive with SOVERSION bumps. 

Moving forward now that JsonCpp is squarely in the maintainence phase of its lifecycle, I think we should avoid doing any SOVERSION bumps at all and I want the project to more carefully keep the SOVERSION the same as much as possible.

Ideally, SOVERSION 27 will be the last SOVERSION for JsonCpp, at least in the near future.

**Comment by jengelh** [comment]
In the specific case of the switch from 24 to 25, the ABI *did* change [git diff 1.9.4..1.9.5] in at least one place:

```c++
diff --git include/json/value.h include/json/value.h
index df1eba6..0edeb05 100644
--- include/json/value.h
+++ include/json/value.h
@@ -918,8 +918,8 @@ public:
    *  because the returned references/pointers can be used
    *  to change state of the base class.
    */
-  reference operator*() { return deref(); }
-  pointer operator->() { return &deref(); }
+  reference operator*() const { return const_cast<reference>(deref()); }
+  pointer operator->() const { return const_cast<pointer>(&deref()); }
 };
 
 inline void swap(Value& a, Value& b) { a.swap(b); }
```
